hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/throttle/PressureAwareCompactionThroughputController.java

```diff
@@ -146,4 +146,38 @@ public String toString() {
         + throughputDesc(getMaxThroughput()) + ", activeCompactions=" + activeOperations.size()
         + "]";
   }
+
+  @Override
+  public void updateConfig(Configuration newConf,
+      final RegionServerServices server) {
+    LOG.debug("update Config");
+    this.setConf(newConf);
+    restartChore(server);
+
+  }
+
+  public void restartChore(RegionServerServices server) {
```

Comment:
Does this method need to be public?

Reply:
> Does this method need to be public?

It should be changed to private.

Reply:
‘So this means we will never actually stop a controller?’
@Apache9 
Because when we execute 'update_config', the running compaction is still use the old Controller object.
So I cache the controller in a map.keep there is always only one PressureAwareCompactionThroughputController object in RS.

If we stop controller after 'update_config' everytime,the running compaction still use the old controller object,and new compaction will use the new controller object.
the total higher.bound,lower.bound maybe more than we expected.